.github/workflows/ci_workflows.yml

```diff
@@ -110,3 +108,34 @@ jobs:
         - name: (Allowed Failure) Python 3.12 with remote data and dev version of key dependencies
           linux: py312-test-devdeps
           posargs: --remote-data=any --verbose
+
+  report_diff_test_coverage:
+    needs: [ tests ]
+    if: always() && github.event_name == 'pull_request'
+    name: report diff test coverage
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3
+      - run: pip install diff-cover coverage
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - run: pip install .
+      - uses: actions/download-artifact@v4
+        with:
+          pattern: ".coverage*"
+          merge-multiple: true
+      - run: |
+          [ "$(ls -A .coverage*)" ] && exit 0 || exit 1
+      - run: coverage xml
+      - run: diff-cover coverage.xml --compare-branch origin/${{ github.base_ref }} --markdown-report diff_coverage.md
+      - run: cat diff_coverage.md >> $GITHUB_STEP_SUMMARY
+      - if: github.event_name == 'pull_request_target'
+        uses: thollander/actions-comment-pull-request@v2.5.0
```

Comment:
I don't see why we need yet another third-party action here. We can use `actions/github-script` directly like this?

https://github.com/astropy/astropy/blob/0d33f688aa1d83b520642ce2ba1302e3ab69afd5/.github/workflows/open_actions.yml#L95

Also, why does this part have to be only in `astropy` repo and not upstream over at OpenAstronomy?

Reply:
this job should only run when triggered by a pull request; however the `github.event` visible inside the OpenAstronomy workflow is `workflow_call`, because it's called by another workflow. I experimented with adding extra inputs to the OpenAstronomy workflow (such as a bool `is-pull-request` or a string `compare-diff-coverage-against` of the base branch); however, that adds a lot of complication to the central workflow so I figured it would be easier to do it downstream

Reply:
Why that particular one though? For example, https://github.com/marketplace/actions/create-or-update-comment has more stars and https://github.com/peter-evans/create-or-update-comment is more recently active (though the release is a month behind the other one).

Reply:
Oh! I only looked at two or so. We can definitely use one with more stars / more recent